src/main/java/com/dougnoel/sentinel/steps/APISteps.java

```diff
@@ -150,37 +150,65 @@ public static void verifyResponseTime(double time) {
-        boolean partialMatch = matchType.contains("contain");
+	@Then("^I validate the response( does not)? (has|have|contains?) the text (.*?)$")
+	public static void verifyResponseContains(String assertion, String matchType, String text) {
+		boolean negate = !StringUtils.isEmpty(assertion);
+		boolean partialMatch = matchType.contains("contain");
 
-        int responseCode = APIManager.getResponse().getResponseCode();
-        String responseText = APIManager.getResponse().getResponse();
-        String expectedResult = SentinelStringUtils.format(
-                "Expected the response to {}{} the text {}. The response had a response code of {} and contained the text: {}",
-                (negate ? "not " : ""), (partialMatch ? "contain" : "exactly match"), text, responseCode, responseText
-                        .replace("\n", " "));
-        log.trace(expectedResult);
-        if (partialMatch) {
-            if (negate) {
-                assertFalse(expectedResult, responseText.contains(text));
-            } else {
-                assertTrue(expectedResult, responseText.contains(text));
-            }
-        } else {
-            if (negate) {
-                assertFalse(expectedResult, StringUtils.equals(responseText, text));
-            } else {
-                assertTrue(expectedResult, StringUtils.equals(responseText, text));
-            }
-        }
-    }
+		int responseCode = APIManager.getResponse().getResponseCode();
+		String responseText = APIManager.getResponse().getResponse();
+		String expectedResult = SentinelStringUtils.format(
+			"Expected the response to {}{} the text {}. The response had a response code of {} and contained the text: {}",
+			(negate ? "not " : ""), (partialMatch ? "contain" : "exactly match"), text, responseCode, responseText
+				.replace("\n", " "));
+		log.trace(expectedResult);
+
+		boolean negateResult;
+		boolean result;
+		boolean textStartEndQuotes = false;
+		String startEndQuoteStripText = text;
+		if(text.matches("^\".*?\"$")) {
+			textStartEndQuotes = true;
+			startEndQuoteStripText = text.substring(1, text.length() - 1);
+		}
+
+		if (partialMatch) {
+			if(textStartEndQuotes) {
+				result = responseText.contains(startEndQuoteStripText) || responseText.contains(text);
+				negateResult = responseText.contains(startEndQuoteStripText) && responseText.contains(text);
+			}
+			else {
+				result = responseText.contains(text);
+				negateResult = result;
+			}
+		} else {
+			if(textStartEndQuotes) {
+				result = StringUtils.equals(responseText, text) || StringUtils.equals(responseText, startEndQuoteStripText);
+				negateResult = StringUtils.equals(responseText, text) && StringUtils.equals(responseText, startEndQuoteStripText);
+			}
+			else {
+				result = responseText.contains(text);
+				negateResult = result;
+			}
+		}
+
+		if (negate)
+			assertFalse(expectedResult, negateResult);
+		else
+			assertTrue(expectedResult, result);
```

Comment:
We had a miscommunictaion. This is what I was attempting to describe:

```suggestion
if (partialMatch) {
            if (negate) {
                assertTrue(expectedResult, SentinelStringUtils.doesNotContain(responseText, text));
            } else {
                assertTrue(expectedResult, SentinelStringUtils.contains(responseText, text));
            }
        } else {
            if (negate) {
                assertTrue(expectedResult, SentinelStringUtils.doesNotequal(responseText, text));
            } else {
                assertTrue(expectedResult, SentinelStringUtils.equals(responseText, text));
            }
        }
```

Those four methods need to be created in SentinelStringUtils, based on our conversations. Then we can reuse those methods in our other text verification areas.

Reply:
Same logic at least, but yes Turch and I came to this conclusion. Though where to put it we weren't entirely sure so this helps a lot.